### PR TITLE
[Archetype] Removed Archetype component and bundle

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

### CODE REVIEW

**Comment by tchapi** [comment]
Since ArchetypeBundle is (_will be_) removed from Core, if we need it anyway, how do we install it back ? I would guess something along the lines of :

    composer require sylius/archetype-bundle 1.0

.. is it all it will take to be on par with what we have now in Core or is there also some other things to do ?

thanks a lot !

**Comment by adamelso** [comment]
@tchapi I can continue to support this package (since I created it) making it a separate package but changing the namespace from Sylius. Whatever works best for the Sylius team and everyone else still using archetypes.

**Comment by tchapi** [comment]
Oh, I see. You mean Sylius will not use archetypes at all in the coming versions ? That's a pity... but if you still continue to support it I'm happy then. I think many users use archetypes to create new types of Products.

Whenever you have separated it and have the new namespace, let me know ! So I can already integrate it in my project in lieu of the Sylius one

Thanks Adam

File: docs/components/Product/basic_usage.rst [hunk 1]
```diff
@@ -150,6 +127,6 @@ Product options management
    $product->getOptions(); // Returns an array containing all inserted options.
 
 .. note::
-   Same as in **Archetype** case, you can use **Option** objects from this component,
+   Doesn't matter if you use **Option** objects from this component,
```
Comment (michalmarcinkowski) [inline docs/components/Product/basic_usage.rst]: `Doesn't matter`? I would write `You can use Option objects ....`

**Comment by pjedrzejewski** [comment]
Thank you Jasiek! :)

**Comment by pjedrzejewski** [comment]
@adamelso Hey Adam! The subtree split repo stays in place, but I will be happy to "donate" the bundle to anyone who wants to maintain it. This kind of bundle/functionality may have a place in future versions of Sylius core, but for now it could be totally standalone. Let me know if you want to take care of it. Alternative is reimplementing the functionality in a new bundle, which does not provide generic archetype functionality (which is costly to maintain) but adds this only for Sylius-case, which is much simpler option.

**Comment by inspiran** [comment]
@pjedrzejewski Maybe consider to found a FriendsofSylius kind of organization where you can move sylius-descoped bundles  into.

**Comment by adamelso** [comment]
@inspiran Funny, I had exactly the same idea 👍  I do think important that Sylius have some sort of ecosystem from the start.

@pjedrzejewski I'm happy to take ownership of this component, better yet if part of a wider "FriendsOfSylius" organization.

**Comment by coox** [comment]
FWIW, here is how to use product archetypes the same way they were provided by `sylius/sylius` before their [removal](https://github.com/Sylius/Sylius/commit/de70496aff4ddb56d5d00448dec6daeb56b6f8c0).

- Add a dependency to [sylius/archetype-bundle](https://packagist.org/packages/sylius/archetype-bundle) in your [Composer](https://getcomposer.org/) configuration, with minimum stability matching the one of your Sylius requirement.

- Define the following parameters (typically in your `app/config.yml`):
```yml
sylius_archetype:
    resources:
        product:
            archetype:
                classes:
                    model: Sylius\Component\Archetype\Model\Archetype
                    interface: Sylius\Component\Archetype\Model\ArchetypeInterface
                translation:
                    classes:
                        model: Sylius\Component\Archetype\Model\ArchetypeTranslation
                        interface: Sylius\Component\Archetype\Model\ArchetypeTranslationInterface
            subject: Sylius\Component\Core\Model\Product
            attribute: Sylius\Component\Product\Model\Attribute
            option: Sylius\Component\Product\Model\Option
```

With this configuration, you can enjoy `sylius.repository.product_archetype` and other goodies that were previously included in core.

**Comment by MatthieuG9** [comment]
Please, someone can explain me why sylius doesn't support archetype anymore ?
Does what mean that sylius project will no more support different kind of products ? For example books with an "author" attribute and t-shirt with "size" attribute.

**Comment by Zales0123** [comment]
@MatthieuG9 *Sylius* is heading to alpha and then beta release. Due to that fact, *Sylius* just can't contains all awesome features it had, as they all requires big effort to maintain and improve. Most of them will be probably reattached to *Sylius* core, but for now we must focus on fixing bugs and stabilizing, which sometimes requires cutting off some features. For more information, please look at the [Sylius roadmap blogpost](http://sylius.org/blog/sylius-roadmap-update). 
Moreover, there are still variants and attributes in *Sylius*, which can allow differentiating your products pretty well ;)

**Comment by tchapi** [comment]
How do we do in this case : https://github.com/Sylius/Sylius/issues/6221 ?

**Pull request merged**